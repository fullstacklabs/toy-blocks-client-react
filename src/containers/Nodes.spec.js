import React from "react";
import { shallow } from "enzyme";
import configureMockStore from "redux-mock-store";
import thunk from 'redux-thunk';
import { Provider } from "react-redux";
import { create } from "react-test-renderer";
import ConnectedNodes, { Nodes } from "./Nodes";
import Node from "../components/Node";

describe("<Nodes />", () => {
  const actions = {
    checkNodeStatuses: jest.fn()
  };

  const nodes = {
    list: [
      {
        url: 'https://thawing-springs-53971.herokuapp.com',
        online: false,
        name: 'Node 1',
        loading: false
      },
      {
        url: 'https://secret-lowlands-62331.herokuapp.com',
        online: false,
        name: 'Node 2',
        loading: false
      }
    ]
  };

  const nodesLoading = {
    list: [
      {
        url: 'https://thawing-springs-53971.herokuapp.com',
        online: false,
        name: 'Node 1',
        loading: true
      },
      {
        url: 'https://secret-lowlands-62331.herokuapp.com',
        online: false,
        name: 'Node 2',
        loading: true
      }
    ]
  }

  const nodesOnlineFetching = {
    list: [
      {
        url: 'https://thawing-springs-53971.herokuapp.com',
        online: true,
        name: 'Node 1',
        loading: false,
        fetching: true
      },
      {
        url: 'https://secret-lowlands-62331.herokuapp.com',
        online: true,
        name: 'Node 2',
        loading: false,
        fetching: true
      }
    ]
  }

  const nodesBlock = {
    list: [
      {
        url: 'https://thawing-springs-53971.herokuapp.com',
        online: true,
        name: 'Node 1',
        loading: false,
        fetching: false,
        blocks: [
          { id: 2, attributes: { data: 'hi i am data2 '} },
          { id: 3, attributes: { data: 'hi i am data definetly data '} }
        ]
      },
      {
        url: 'https://secret-lowlands-62331.herokuapp.com',
        online: true,
        name: 'Node 2',
        loading: false,
        fetching: false,
        blocks: [
          { id: 1, attributes: { data: 'hi i am data '} }
        ]
      }
    ]
  }

  it("should contain <Node />", () => {
    const wrapper = shallow(
      <Nodes
        actions={actions}
        nodes={nodes}
      />
    );

    expect(wrapper.find(Node).length).toEqual(2);
  });

  it("should match default snapshot", () => {
    const middlewares = [thunk];
    const store = configureMockStore(middlewares)({nodes});
    const component = create(
      <Provider store={store}>
        <ConnectedNodes />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should match loading snapshot", () => {
    const middlewares = [thunk];
    const store = configureMockStore(middlewares)({nodes: nodesLoading});
    const component = create(
      <Provider store={store}>
        <ConnectedNodes />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
  
  it("should match fetching snapshot", () => {
    const middlewares = [thunk];
    const store = configureMockStore(middlewares)({ nodes: nodesOnlineFetching});
    const component = create(
      <Provider store={store}>
        <ConnectedNodes />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("should match loaded snapshot", () => {
    const middlewares = [thunk];
    const store = configureMockStore(middlewares)({ nodes: nodesBlock});
    const component = create(
      <Provider store={store}>
        <ConnectedNodes />
      </Provider>
    );
    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });

});
