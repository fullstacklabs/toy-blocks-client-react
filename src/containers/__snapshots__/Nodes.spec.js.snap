// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nodes /> should match default snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-2"
>
  <h1
    className="MuiTypography-root MuiTypography-h4"
  >
    <strong
      style={
        Object {
          "color": "#000",
        }
      }
    >
      Nodes
    </strong>
  </h1>
  <div
    className="MuiPaper-root MuiExpansionPanel-root makeStyles-root-33 MuiExpansionPanel-rounded MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root makeStyles-summary-34"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content makeStyles-content-37"
      >
        <div
          className="MuiBox-root MuiBox-root-83 makeStyles-summaryContent-35"
        >
          <div
            className="MuiBox-root MuiBox-root-84"
          >
            <h5
              className="MuiTypography-root makeStyles-heading-40 MuiTypography-h5"
            >
              Node 1
            </h5>
            <h6
              className="MuiTypography-root makeStyles-secondaryHeading-41 MuiTypography-subtitle1"
            >
              https://thawing-springs-53971.herokuapp.com
            </h6>
          </div>
          <div
            className="MuiBox-root MuiBox-root-89"
          >
            <span
              className="makeStyles-dot-85 makeStyles-dot-87"
            />
            <span
              className="makeStyles-text-86 makeStyles-text-88"
            >
              OFFLINE
            </span>
          </div>
        </div>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon makeStyles-icon-36 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onClick={undefined}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            color={undefined}
            focusable="false"
            role={undefined}
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby={undefined}
            id={undefined}
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root makeStyles-details-39"
            >
              <div
                className="MuiBox-root MuiBox-root-119"
              >
                <p
                  className="MuiTypography-root makeStyles-loading-118 MuiTypography-body1"
                >
                  Offline, couldn't retrieve blocks
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiExpansionPanel-root makeStyles-root-33 MuiExpansionPanel-rounded MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root makeStyles-summary-34"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content makeStyles-content-37"
      >
        <div
          className="MuiBox-root MuiBox-root-120 makeStyles-summaryContent-35"
        >
          <div
            className="MuiBox-root MuiBox-root-121"
          >
            <h5
              className="MuiTypography-root makeStyles-heading-40 MuiTypography-h5"
            >
              Node 2
            </h5>
            <h6
              className="MuiTypography-root makeStyles-secondaryHeading-41 MuiTypography-subtitle1"
            >
              https://secret-lowlands-62331.herokuapp.com
            </h6>
          </div>
          <div
            className="MuiBox-root MuiBox-root-124"
          >
            <span
              className="makeStyles-dot-85 makeStyles-dot-122"
            />
            <span
              className="makeStyles-text-86 makeStyles-text-123"
            >
              OFFLINE
            </span>
          </div>
        </div>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon makeStyles-icon-36 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onClick={undefined}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            color={undefined}
            focusable="false"
            role={undefined}
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby={undefined}
            id={undefined}
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root makeStyles-details-39"
            >
              <div
                className="MuiBox-root MuiBox-root-125"
              >
                <p
                  className="MuiTypography-root makeStyles-loading-118 MuiTypography-body1"
                >
                  Offline, couldn't retrieve blocks
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Nodes /> should match fetching snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-149"
>
  <h1
    className="MuiTypography-root MuiTypography-h4"
  >
    <strong
      style={
        Object {
          "color": "#000",
        }
      }
    >
      Nodes
    </strong>
  </h1>
  <div
    className="MuiPaper-root MuiExpansionPanel-root makeStyles-root-33 MuiExpansionPanel-rounded MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root makeStyles-summary-34"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content makeStyles-content-37"
      >
        <div
          className="MuiBox-root MuiBox-root-150 makeStyles-summaryContent-35"
        >
          <div
            className="MuiBox-root MuiBox-root-151"
          >
            <h5
              className="MuiTypography-root makeStyles-heading-40 MuiTypography-h5"
            >
              Node 1
            </h5>
            <h6
              className="MuiTypography-root makeStyles-secondaryHeading-41 MuiTypography-subtitle1"
            >
              https://thawing-springs-53971.herokuapp.com
            </h6>
          </div>
          <div
            className="MuiBox-root MuiBox-root-154"
          >
            <span
              className="makeStyles-dot-85 makeStyles-dot-152"
            />
            <span
              className="makeStyles-text-86 makeStyles-text-153"
            >
              ONLINE
            </span>
          </div>
        </div>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon makeStyles-icon-36 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onClick={undefined}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            color={undefined}
            focusable="false"
            role={undefined}
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby={undefined}
            id={undefined}
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root makeStyles-details-39"
            >
              <div
                className="MuiBox-root MuiBox-root-155"
              >
                <p
                  className="MuiTypography-root makeStyles-loading-118 MuiTypography-body1"
                >
                  Fetching...
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiExpansionPanel-root makeStyles-root-33 MuiExpansionPanel-rounded MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root makeStyles-summary-34"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content makeStyles-content-37"
      >
        <div
          className="MuiBox-root MuiBox-root-156 makeStyles-summaryContent-35"
        >
          <div
            className="MuiBox-root MuiBox-root-157"
          >
            <h5
              className="MuiTypography-root makeStyles-heading-40 MuiTypography-h5"
            >
              Node 2
            </h5>
            <h6
              className="MuiTypography-root makeStyles-secondaryHeading-41 MuiTypography-subtitle1"
            >
              https://secret-lowlands-62331.herokuapp.com
            </h6>
          </div>
          <div
            className="MuiBox-root MuiBox-root-160"
          >
            <span
              className="makeStyles-dot-85 makeStyles-dot-158"
            />
            <span
              className="makeStyles-text-86 makeStyles-text-159"
            >
              ONLINE
            </span>
          </div>
        </div>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon makeStyles-icon-36 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onClick={undefined}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            color={undefined}
            focusable="false"
            role={undefined}
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby={undefined}
            id={undefined}
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root makeStyles-details-39"
            >
              <div
                className="MuiBox-root MuiBox-root-161"
              >
                <p
                  className="MuiTypography-root makeStyles-loading-118 MuiTypography-body1"
                >
                  Fetching...
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Nodes /> should match loaded snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-162"
>
  <h1
    className="MuiTypography-root MuiTypography-h4"
  >
    <strong
      style={
        Object {
          "color": "#000",
        }
      }
    >
      Nodes
    </strong>
  </h1>
  <div
    className="MuiPaper-root MuiExpansionPanel-root makeStyles-root-33 MuiExpansionPanel-rounded MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root makeStyles-summary-34"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content makeStyles-content-37"
      >
        <div
          className="MuiBox-root MuiBox-root-163 makeStyles-summaryContent-35"
        >
          <div
            className="MuiBox-root MuiBox-root-164"
          >
            <h5
              className="MuiTypography-root makeStyles-heading-40 MuiTypography-h5"
            >
              Node 1
            </h5>
            <h6
              className="MuiTypography-root makeStyles-secondaryHeading-41 MuiTypography-subtitle1"
            >
              https://thawing-springs-53971.herokuapp.com
            </h6>
          </div>
          <div
            className="MuiBox-root MuiBox-root-167"
          >
            <span
              className="makeStyles-dot-85 makeStyles-dot-165"
            />
            <span
              className="makeStyles-text-86 makeStyles-text-166"
            >
              ONLINE
            </span>
          </div>
        </div>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon makeStyles-icon-36 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onClick={undefined}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            color={undefined}
            focusable="false"
            role={undefined}
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby={undefined}
            id={undefined}
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root makeStyles-details-39"
            >
              <div
                className="MuiBox-root MuiBox-root-168 makeStyles-container-114"
              >
                <div
                  className="MuiBox-root MuiBox-root-169 makeStyles-block-117"
                >
                  <span
                    className="makeStyles-heading-116"
                  >
                    002
                  </span>
                  <span
                    className="makeStyles-text-115"
                  >
                    hi i am data2 
                  </span>
                </div>
                <div
                  className="MuiBox-root MuiBox-root-170 makeStyles-block-117"
                >
                  <span
                    className="makeStyles-heading-116"
                  >
                    003
                  </span>
                  <span
                    className="makeStyles-text-115"
                  >
                    hi i am data definetly data 
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiExpansionPanel-root makeStyles-root-33 MuiExpansionPanel-rounded MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root makeStyles-summary-34"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content makeStyles-content-37"
      >
        <div
          className="MuiBox-root MuiBox-root-171 makeStyles-summaryContent-35"
        >
          <div
            className="MuiBox-root MuiBox-root-172"
          >
            <h5
              className="MuiTypography-root makeStyles-heading-40 MuiTypography-h5"
            >
              Node 2
            </h5>
            <h6
              className="MuiTypography-root makeStyles-secondaryHeading-41 MuiTypography-subtitle1"
            >
              https://secret-lowlands-62331.herokuapp.com
            </h6>
          </div>
          <div
            className="MuiBox-root MuiBox-root-175"
          >
            <span
              className="makeStyles-dot-85 makeStyles-dot-173"
            />
            <span
              className="makeStyles-text-86 makeStyles-text-174"
            >
              ONLINE
            </span>
          </div>
        </div>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon makeStyles-icon-36 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onClick={undefined}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            color={undefined}
            focusable="false"
            role={undefined}
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby={undefined}
            id={undefined}
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root makeStyles-details-39"
            >
              <div
                className="MuiBox-root MuiBox-root-176 makeStyles-container-114"
              >
                <div
                  className="MuiBox-root MuiBox-root-177 makeStyles-block-117"
                >
                  <span
                    className="makeStyles-heading-116"
                  >
                    001
                  </span>
                  <span
                    className="makeStyles-text-115"
                  >
                    hi i am data 
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Nodes /> should match loading snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-136"
>
  <h1
    className="MuiTypography-root MuiTypography-h4"
  >
    <strong
      style={
        Object {
          "color": "#000",
        }
      }
    >
      Nodes
    </strong>
  </h1>
  <div
    className="MuiPaper-root MuiExpansionPanel-root makeStyles-root-33 MuiExpansionPanel-rounded MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root makeStyles-summary-34"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content makeStyles-content-37"
      >
        <div
          className="MuiBox-root MuiBox-root-137 makeStyles-summaryContent-35"
        >
          <div
            className="MuiBox-root MuiBox-root-138"
          >
            <h5
              className="MuiTypography-root makeStyles-heading-40 MuiTypography-h5"
            >
              Node 1
            </h5>
            <h6
              className="MuiTypography-root makeStyles-secondaryHeading-41 MuiTypography-subtitle1"
            >
              https://thawing-springs-53971.herokuapp.com
            </h6>
          </div>
          <div
            className="MuiBox-root MuiBox-root-141"
          >
            <span
              className="makeStyles-dot-85 makeStyles-dot-139"
            />
            <span
              className="makeStyles-text-86 makeStyles-text-140"
            >
              LOADING
            </span>
          </div>
        </div>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon makeStyles-icon-36 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onClick={undefined}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            color={undefined}
            focusable="false"
            role={undefined}
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby={undefined}
            id={undefined}
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root makeStyles-details-39"
            >
              <div
                className="MuiBox-root MuiBox-root-142"
              >
                <p
                  className="MuiTypography-root makeStyles-loading-118 MuiTypography-body1"
                >
                  Loading...
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiPaper-root MuiExpansionPanel-root makeStyles-root-33 MuiExpansionPanel-rounded MuiPaper-elevation3 MuiPaper-rounded"
  >
    <div
      aria-disabled={false}
      aria-expanded={false}
      className="MuiButtonBase-root MuiExpansionPanelSummary-root makeStyles-summary-34"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <div
        className="MuiExpansionPanelSummary-content makeStyles-content-37"
      >
        <div
          className="MuiBox-root MuiBox-root-143 makeStyles-summaryContent-35"
        >
          <div
            className="MuiBox-root MuiBox-root-144"
          >
            <h5
              className="MuiTypography-root makeStyles-heading-40 MuiTypography-h5"
            >
              Node 2
            </h5>
            <h6
              className="MuiTypography-root makeStyles-secondaryHeading-41 MuiTypography-subtitle1"
            >
              https://secret-lowlands-62331.herokuapp.com
            </h6>
          </div>
          <div
            className="MuiBox-root MuiBox-root-147"
          >
            <span
              className="makeStyles-dot-85 makeStyles-dot-145"
            />
            <span
              className="makeStyles-text-86 makeStyles-text-146"
            >
              LOADING
            </span>
          </div>
        </div>
      </div>
      <div
        aria-disabled={false}
        aria-hidden={true}
        className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon makeStyles-icon-36 MuiIconButton-edgeEnd"
        onBlur={[Function]}
        onClick={undefined}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role={null}
        tabIndex={null}
      >
        <span
          className="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            color={undefined}
            focusable="false"
            role={undefined}
            viewBox="0 0 24 24"
          >
            <path
              d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            />
          </svg>
        </span>
      </div>
    </div>
    <div
      className="MuiCollapse-container MuiCollapse-hidden"
      style={
        Object {
          "minHeight": "0px",
        }
      }
    >
      <div
        className="MuiCollapse-wrapper"
      >
        <div
          className="MuiCollapse-wrapperInner"
        >
          <div
            aria-labelledby={undefined}
            id={undefined}
            role="region"
          >
            <div
              className="MuiExpansionPanelDetails-root makeStyles-details-39"
            >
              <div
                className="MuiBox-root MuiBox-root-148"
              >
                <p
                  className="MuiTypography-root makeStyles-loading-118 MuiTypography-body1"
                >
                  Loading...
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
